packages/hash/frontend/src/components/hooks/useUpdateAuthenticatedUser.ts

```diff
@@ -0,0 +1,117 @@
+import { useMutation } from "@apollo/client";
+import { useCallback, useState } from "react";
+
+import { types } from "@hashintel/hash-shared/types";
+import { extractBaseUri } from "@blockprotocol/type-system-web";
+import { GraphQLError } from "graphql";
+import {
+  UpdatePersistedEntityMutation,
+  UpdatePersistedEntityMutationVariables,
+} from "../../graphql/apiTypes.gen";
+import { mustGetEntity, Subgraph } from "../../lib/subgraph";
+import { AuthenticatedUser, constructAuthenticatedUser } from "../../lib/user";
+import { updatePersistedEntityMutation } from "../../graphql/queries/knowledge/entity.queries";
+import { useAuthenticatedUser } from "./useAuthenticatedUser";
+
+type UpdateAuthenticatedUserParams = {
+  shortname?: string;
+  preferredName?: string;
+};
+
+export const useUpdateAuthenticatedUser = () => {
+  const { authenticatedUser, refetch } = useAuthenticatedUser();
+
+  const [updatePersistedEntity] = useMutation<
+    UpdatePersistedEntityMutation,
+    UpdatePersistedEntityMutationVariables
+  >(updatePersistedEntityMutation, { errorPolicy: "all" });
```

Comment:
I don't think partial data would be returned here, I chose this error policy so that the consumer of this hook can easily make use of the returned error messages without having to wrap a function call in a `try { ... } catch { ... }`

Reply:
This feels like one of those decisions that hasn't been appropriately captured. Do we want an Apollo style guide or something in the repo? Like if we're going to use this here purely to stop a try catch block, surely we want to use it on every mutation? I don't want to be digging around the error policy of the mutation to find out how to catch my errors.